src/orchard/info.clj

```diff
@@ -63,30 +63,34 @@
 
 (defn clj-meta
   {:added "0.5"}
-  [{:keys [dialect ns sym computed-ns unqualified-sym]}]
+  [{:keys [dialect ns sym computed-ns unqualified-sym] :as opts}]
   {:pre [(= dialect :clj)]}
   (let [ns (or ns computed-ns)]
-    (or
-     ;; it's a special (special-symbol?)
-     (m/special-sym-meta sym)
-     ;; it's a var
-     (some-> ns (m/resolve-var sym) (m/var-meta))
-     ;; it's a Java constructor/static member symbol
-     (some-> ns (java/resolve-symbol sym))
-     ;; it's an unqualified sym maybe referred
-     (some-> ns (m/resolve-var unqualified-sym) (m/var-meta))
-     ;; it's a Java class/record type symbol
-     (some-> ns (java/resolve-type unqualified-sym))
-     ;; it's an alias for another ns
-     (some-> ns (m/resolve-aliases) (get sym) (m/ns-meta))
-     ;; We use :unqualified-sym *exclusively* here because because our :ns is
-     ;; too ambiguous.
-     ;;
-     ;; Observe the incorrect behavior (should return nil, there is a test):
-     ;;
-     ;;   (info '{:ns clojure.core :sym non-existing}) ;;=> {:author "Rich Hickey" :ns clojure.core ...}
-     ;;
-     (some-> (find-ns unqualified-sym) (m/ns-meta)))))
+    (if (find-ns ns)
+      (or
+       ;; it's a special (special-symbol?)
+       (m/special-sym-meta sym)
+       ;; it's a var
+       (some-> ns (m/resolve-var sym) (m/var-meta))
+       ;; it's a Java constructor/static member symbol
+       (some-> ns (java/resolve-symbol sym))
+       ;; it's an unqualified sym maybe referred
+       (some-> ns (m/resolve-var unqualified-sym) (m/var-meta))
+       ;; it's a Java class/record type symbol
+       (some-> ns (java/resolve-type unqualified-sym))
+       ;; it's an alias for another ns
+       (some-> ns (m/resolve-aliases) (get sym) (m/ns-meta))
+       ;; We use :unqualified-sym *exclusively* here because because our :ns is
+       ;; too ambiguous.
+       ;;
+       ;; Observe the incorrect behavior (should return nil, there is a test):
+       ;;
+       ;;   (info '{:ns clojure.core :sym non-existing}) ;;=> {:author "Rich Hickey" :ns clojure.core ...}
+       ;;
+       (some-> (find-ns unqualified-sym) (m/ns-meta)))
+      ;; Lookups in files whose namespaces don't exist yet should still be able
+      ;; to resolve built-ins and fully-qualified syms
+      (recur (assoc opts :ns 'clojure.core)))))
```

Comment:
Very minor, but the large distance between the `if` and this branch makes it a bit hard to track the control flow.

By using an `if-not` instead one gets these "degenerate cases" out of one's way, similarly to https://wiki.c2.com/?GuardClause

Reply:
Thanks for the feedback. I've addressed this one. I'll still be adding a test and changelog etc.